apps/srt-live-transmit.cpp

```diff
@@ -253,11 +253,12 @@ int main( int argc, char** argv )
         cerr << "\t-s:<stats-report-freq=0> - frequency of status report\n";
         cerr << "\t-k - crash on error (aka developer mode)\n";
         cerr << "\t-v - verbose mode (prints also size of every data packet passed)\n";
+        cerr << "\t-f - full counters (prints total statistics without reset)\n";
         return 1;
     }
 
     int timeout = stoi(Option("30", "t", "to", "timeout"), 0, 0);
-    size_t chunk = stoul(Option("0", "c", "chunk"), 0, 0);
+    unsigned long chunk = stoul(Option("0", "c", "chunk"), 0, 0);
```

Comment:
What problem do you have with `size_t`? This is usually an alias to `unsigned long`, except 64-bit platforms where it's an alias to `unsigned long long`. I don't think it clears any signed comparison warnings.

Reply:
Hi @ethouris ,

>  This is usually an alias to unsigned long

Sorry? `size_t` it's the size of the Object Pointer? So, if you're not using Objects, then it's a good practice to use standard number types. If you don't do it like that, then you can have troubles with other platforms (x86 vs ARM or Mips, 32/64 bit, Endianess, etc.).

Futhermore, I suggest to use `unsigned long` for **counters** as they can have a very long life (if not cleared), or `unsigned long long` if you prefer it.

Reply:
The `std::size_t` type is officially the result of `sizeof` operator, and possibly plays also other roles in the standard library. It's theoretically part of the standard library, but also bound to language statements, so it's by all means a "standard number type".

Reply:
Hi @ethouris ,

If you read the documentation: `typedef /*implementation-defined*/ size_t;`.
So, if you don't like to have troubles with interoperatibility using different platforms, then you need to use **explicit** known types. My suggestion it's to not use at all `size_t` except when you really need it.

For example, the reason to change the type in this branch it's because the use of `std::stoul`. If you compile for another platorm then the type can change, and you can finish with an error.

Please, believe me. To compile this library on multiple platforms, we need to go away of `size_t`.

Reply:
Let me split my meaning into two parts.
1. Not sure what this **explicit** should mean. If that not builtin - it's still part of the language standard, which mandates it to be an unsigned integer type, just unspecified as to which one. If that it's not fixed-size type, well, `unsigned long` has also an implementation-defined size. And it does differ on 32 and 64-bit platforms.
2. On the second thought, maybe this is better. I had some problems with `size_t` on 64-bit platforms which were caused by a not-well-understood rules of extending shorter types into longer ones. In particular, I tried to use `-1u` to make an integer value consisting of only set bits. This is one of just a few corner cases. Even though that, I think sticking with `unsigned long` is better at least for the logics of the solution. Honestly, I used it because it's shorter.

So, leave it as is, I don't have a problem with it.

Reply:
:+1: